tuf/store/interfaces.go

```diff
@@ -12,7 +17,8 @@ type MetadataStore interface {
 
 // PublicKeyStore must be implemented by a key service
 type PublicKeyStore interface {
-	GetKey(role string) ([]byte, error)
+	GetKey(role string) (data.PublicKey, error)
+	RotateKey(role string, cs signed.CryptoService, roots ...data.PublicKey) (data.PublicKey, error)
```

Comment:
Hm, so I can see HTTPStore being just `[]bytes`, but do you think the interface should also just be `[]bytes`?  There can be something else between `HTTPStore` and the client, and `HTTPStore` doesn't have to definitely implement the interface.  Alternately, maybe `HTTPStore` can just implement the metadata part and something else that wraps it can implement the signing/unmarshalling?